### ISSUE TITLE
Failing to recreate the ad-hoc provisioning profiles after a new device is added to the portal

### ISSUE THREAD

**Issue by sirghi** [fastlane/fastlane#12552]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

I have a question about **match**. On my project I am maintaining the cert repo.

Recently a new device UDID was added to the portal, and I tried to recreate the ad-hoc profiles using `readonly:false, force_for_new_devices:true`(I am an admin on the dev portal). And got the below error:

```
Provisioning profile ‘0b9f8fab-899b-48a7-9bc0-defcdecad0bd’ is not available on the Developer Portal
for the user abc@efg.com
Make sure to use the same user and team every time you run ‘match’ for this
Git repository. This might be caused by deleting the provisioning profile on the Dev Portal
```

Looking at the dev portal, that provisioning profile does exist (and is still valid), and I can see and download it manually. 

Any idea why **match** fails in this case and what can we do?

**Comment by sirghi** [comment]
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.4                                     |
| Ruby                        | 2.2.3                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.rvm/gems/ruby-2.2.3/bin/fastlane         |
| Host                        | Mac OS X 10.13.4 (17E202)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.2.3/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2h  3 May 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.95.0  | 🚫 Update available |


### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                | Version      |
| ------------------ | ------------ |
| slack-notifier     | 2.3.2        |
| multipart-post     | 2.0.0        |
| word_wrap          | 1.0.0        |
| public_suffix      | 2.0.5        |
| tty-screen         | 0.6.4        |
| tty-cursor         | 0.5.0        |
| tty-spinner        | 0.8.0        |
| babosa             | 1.0.2        |
| colored            | 1.2          |
| commander-fastlane | 4.4.6        |
| http-cookie        | 1.0.3        |
| faraday-cookie_jar | 0.0.6        |
| gh_inspector       | 1.1.3        |
| mini_magick        | 4.5.1        |
| multi_xml          | 0.6.0        |
| security           | 0.1.3        |
| declarative-option | 0.1.0        |
| representable      | 3.0.4        |
| mime-types-data    | 3.2016.0521  |
| mime-types         | 3.1          |
| httpclient         | 2.8.3        |
| google-api-client  | 0.13.6       |
| emoji_regex        | 0.1.1        |
| highline           | 1.7.10       |
| excon              | 0.62.0       |
| rubyzip            | 1.2.1        |
| CFPropertyList     | 3.0.0        |
| mini_portile2      | 2.3.0        |
| nokogiri           | 1.8.2        |
| plist              | 3.4.0        |
| atomos             | 0.1.2        |
| claide             | 1.0.2        |
| colored2           | 3.1.2        |
| nanaimo            | 0.2.5        |
| xcodeproj          | 1.5.8        |
| faraday            | 0.15.1       |
| unf                | 0.1.4        |
| domain_name        | 0.5.20180417 |
| faraday_middleware | 0.12.2       |
| fastimage          | 2.1.3        |
| json               | 2.1.0        |
| bundler            | 1.16.1       |
| dotenv             | 2.4.0        |
| psych              | 2.0.8        |
| google_drive       | 2.1.10       |
| os                 | 0.9.6        |
| googleauth         | 0.6.2        |
| signet             | 0.8.1        |
| addressable        | 2.5.2        |
| multi_json         | 1.13.1       |
| jwt                | 2.1.0        |
| memoist            | 0.16.0       |
| retriable          | 3.1.1        |
| uber               | 0.1.0        |
| declarative        | 0.0.10       |
| diffy              | 3.2.0        |
</details>


*generated on:* **2018-05-24**
</details>

**Comment by mosesoak** [comment]
@sirghi we ran into this problem as well with our enterprise cert, not sure if someone messed with something in the apple dev portal to cause it, or if it was an incorrect setup for match.

The way we found match to always work right after adding device was `readonly: false, force: true`. The flag `force` will recreate the provisioning profile even if the device is not being added for the first time. Please reply whether that fixes the issue for you!

Before we found that one of our projects had gotten into the stuck state you posted about. I don't know whether `force:true` would fix it. However, I did find a way to very easily reset the project, and I'm considering making a PR or feature request to Fastlane to remove a project from match, since it's pretty simple:

1. In your match provisioning repo, delete all of the entries for your project, then commit and push this
2. Delete the project's non-shared provisioning profile/s (e.g. delete the development one but not a shared ad hoc or enterprise one)

That made it possible to recreate the project provisioning using `produce` and `match` (w/ `readonly: false`).

**Comment by sirghi** [comment]
Hi @mosesoak 
Thanks for jumping in on this. Unfortunately, neither `readonly: false, force: true` did the trick for us. And in the end we also went with something similar to your second approach: we just deleted the said profile from the repo, pushed, then ran `match` again with `readonly: false`, and it worked.
I do agree, this approach is still rather simple. 
Cheers

But still, hoping we can get more insights into and solve the original issue.
